Remove void spellstone from qb reward and advancements

Enriian opened the pull request:
PR for hiding the Pearl of the Void item from Enigmatic Legacy. Uses a datapack to remove related advancements, a crafttweaker script for hiding the item in JEI, and item is removed as a reward from completing the Wither Eye under Hard Eyes in the QB.

Closes #945

Commit:
Remove void spellstone from qb reward and advancements

Jonah-Hansen reviewed (changes requested):
you'll also need to remove it from the shop page. 

Im thinking we should replace it with something else as a reward though. just not sure what else would be worthy

Commit:
Change wither eye reward from void pearl to curios slots and remove void pearl from shop

Enriian commented:
- Removed Pearl of the Void from the shop.
- Added a choice reward to the wither eye for either a charm or trinket slot (made sure to disable team reward).
- Used arbitrary icons for the trinket and charm as the Curios icons from the inventory were dark and hard to see in the qb.

<img src=https://github.com/user-attachments/assets/06d486ce-73f9-4b7e-9bf7-32db8dcc7e5b width=200/>
<img src=https://github.com/user-attachments/assets/f8d60391-348f-47ea-a397-72b11d21726c width=200/>
<img src=https://github.com/user-attachments/assets/c2fe7a14-a4b4-4b35-b994-e42c5d15c7db width=300/>

Jonah-Hansen reviewed (approved):
code wise this looks good, just wait for approval from merry to make sure this is okay in her opinion
